packages/accounts/src/light-account/account.ts

```diff
@@ -32,45 +37,72 @@ import {
 } from "./utils.js";
 
 export type LightAccount<
+  TEntryPointVersion extends EntryPointVersion,
   TSigner extends SmartAccountSigner = SmartAccountSigner
-> = SmartContractAccountWithSigner<"LightAccount", TSigner> & {
+> = SmartContractAccountWithSigner<
+  TEntryPointVersion,
+  "LightAccount",
+  TSigner
+> & {
   getLightAccountVersion: () => Promise<LightAccountVersion>;
   encodeTransferOwnership: (newOwner: Address) => Hex;
   getOwnerAddress: () => Promise<Address>;
 };
 
 export type CreateLightAccountParams<
+  TEntryPointVersion extends EntryPointVersion,
   TTransport extends Transport = Transport,
   TSigner extends SmartAccountSigner = SmartAccountSigner
 > = Pick<
-  ToSmartContractAccountParams<"LightAccount", TTransport>,
-  "transport" | "chain" | "entryPoint" | "accountAddress"
+  ToSmartContractAccountParams<TEntryPointVersion, "LightAccount", TTransport>,
+  "transport" | "chain"
 > & {
   signer: TSigner;
   salt?: bigint;
+  accountAddress?: Address;
   factoryAddress?: Address;
   initCode?: Hex;
   version?: LightAccountVersion;
-};
+} & EntryPointParameter<TEntryPointVersion>;
 
 export async function createLightAccount<
+  TEntryPointDef extends EntryPointDefRegistry[EntryPointVersion],
+  TEntryPointVersion extends EntryPointVersion = TEntryPointDef extends EntryPointDef<
+    infer U
+  >
+    ? U
+    : never,
   TTransport extends Transport = Transport,
   TSigner extends SmartAccountSigner = SmartAccountSigner
 >(
-  config: CreateLightAccountParams<TTransport, TSigner>
-): Promise<LightAccount<TSigner>>;
+  config: CreateLightAccountParams<TEntryPointVersion, TTransport, TSigner>
+): Promise<LightAccount<TEntryPointVersion, TSigner>>;
 
-export async function createLightAccount({
+export async function createLightAccount<
+  TEntryPointDef extends EntryPointDefRegistry[EntryPointVersion],
+  TEntryPointVersion extends EntryPointVersion = TEntryPointDef extends EntryPointDef<
+    infer U
+  >
+    ? U
+    : never,
+  TTransport extends Transport = Transport,
+  TSigner extends SmartAccountSigner = SmartAccountSigner
+>({
   transport,
   chain,
   signer,
   initCode,
   version = "v1.1.0",
-  entryPoint = getVersion060EntryPoint(chain),
+  entryPoint,
```

Comment:
yea we should just default to 0.6 here and remove the generics for entrypoint def for light account and MA as well. they all require 0.6